File: 3 Array/13 Longest Subarray.py
def longest_subarray(arr, k):
  n = len(arr)
  max = -1
  for i in range(n):
    j = i + 1
    count = arr[i]
    while (count < k and j < n):
      count += arr[j]
      j += 1
    l = j - i
    if (count == k and l > max):
      max = l
  return max

print(longest_subarray([2,5,1,1,3,-1], 5))

File: Array/Problems/13 Longest subarray.cpp
#include <bits/stdc++.h>
using namespace std;

int longestSubarray(int arr[], int n, int k)
{
  int max = -1;
  for (int i = 0; i < (n - 1); i++)
  {
    int j = i + 1;
    int count = arr[i];
    while (count < k && j < n)
    {
      count += arr[j];
      j++;
    }
    int l = (j - i);
    if (count == k && l > max)
      max = l;
    else
      continue;
  }
  return max;
}

int main()
{
  int n;
  cin >> n;
  int arr[n];
  for (int i = 0; i < n; i++)
    cin >> arr[i];
  int k;
  cin >> k;
  int num = longestSubarray(arr, n, k);
  cout << "Longest subarray length: " << num;
  return 0;
}
